src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.InlineContext.cs

```diff
@@ -0,0 +1,465 @@
+                    .ToDictionary(parameterAndName => (ISymbol)parameterAndName.parameterSymbol,
+                        parameterAndName => parameterAndName.name);
+
+                // 1. Make sure after replacing the parameters with the identifier from Caller, there is no variable conflict in callee
+                foreach (var localSymbol in localSymbolsOfCallee)
+                {
+                    var localSymbolName = localSymbol.Name;
+                    while (inUsedNamesOfCallee.Contains(localSymbolName)
+                        || renameTable.ContainsValue(localSymbolName))
+                    {
+                        localSymbolName = GenerateNewName(localSymbolName);
+                    }
+
+                    if (!localSymbolName.Equals(localSymbol.Name))
+                    {
+                        renameTable[localSymbol] = localSymbolName;
+                    }
+                }
+
+                // 2. Make sure no variable conflict after the parameter is moved to caller
+                foreach (var parameterSymbol in parametersNeedGenerateDeclaration)
+                {
+                    var parameterName = parameterSymbol.Name;
+                    while (inUsedNamesOfCaller.Contains(parameterName)
+                        || inUsedNamesOfCallee.Contains(parameterName)
+                        || renameTable.ContainsValue(parameterName))
+                    {
+                        parameterName = GenerateNewName(parameterName);
+                    }
+
+                    if (!parameterName.Equals(parameterSymbol.Name))
+                    {
+                        renameTable[parameterSymbol] = parameterName;
+                    }
+                }
+
+                return renameTable.ToImmutableDictionary();
+            }
+
+            private static SyntaxNode GetInvokingStatement(
+                ISyntaxFacts syntaxFacts, AbstractInlineMethodRefactoringProvider inlineMethodRefactoringProvider, SyntaxNode syntaxNode)
+            {
+                for (var node = syntaxNode; node != null; node = node!.Parent)
+                {
+                    // Is there anything missed here?
+                    if (node != null && (
+                        syntaxFacts.IsLocalDeclarationStatement(node)
+                        || inlineMethodRefactoringProvider.IsEmbeddedStatementOwner(syntaxNode)
+                        || syntaxFacts.IsExpressionStatement(node)))
+                    {
+                        return node;
+                    }
+                }
+
+                return syntaxNode;
+            }
+
+            /// <summary>
+            /// Generate a new identifier name. If <param name="identifierName"/> has a number suffix,
+            /// increase it by 1. Otherwise, append 1 to it.
+            /// </summary>
+            private static string GenerateNewName(string identifierName)
+            {
+                var stack = new Stack<char>();
```

Comment:
there has to be a simpelr and easier way to do this :)

Reply:
I just saw this comment actually.😃 Do you mean there is an existing API for doing this(Found an unused name)?